src/java.base/share/classes/java/time/format/ZoneName.java.template

```diff
@@ -40,26 +40,24 @@ class ZoneName {
 
     public static String toZid(String zid, Locale locale) {
         String mzone = zidToMzone.get(zid);
-        if (mzone == null && aliases.containsKey(zid)) {
-            zid = aliases.get(zid);
-            mzone = zidToMzone.get(zid);
+        if (mzone == null) {
+            String alias = aliases.get(zid);
+            if (alias != null) {
+                zid = alias;
+                mzone = zidToMzone.get(zid);
+            }
         }
         if (mzone != null) {
             Map<String, String> map = mzoneToZidL.get(mzone);
-            if (map != null && map.containsKey(locale.getCountry())) {
-                zid = map.get(locale.getCountry());
-            } else {
+            if (map == null || ((zid = map.get(locale.getCountry())) == null)) {
                 zid = mzoneToZid.get(mzone);
             }
         }
         return toZid(zid);
     }
 
     public static String toZid(String zid) {
-        if (aliases.containsKey(zid)) {
-            return aliases.get(zid);
-        }
-        return zid;
+        return aliases.getOrDefault(zid, zid);
```

Comment:
Is the behavior if zid == null the same?  aliases.getOrDefault will throw NPE on null.
neither Hashmap.containsKey or .get throw on null.

Reply:
>aliases.getOrDefault will throw NPE on null

No, It will not. `aliases` is a HashMap. And HashMap supports null values and keys.

Reply:
Anyway, this method is used only in `DateTimeFormatterBuilder` and it doesn't pass `null` value there.